### ISSUE TITLE
This package should not be responsible of syntax colors

### ISSUE THREAD

**Issue by Nercury** [miqid/atom-language-rust#2]
It is good that that this package defines new styles to differentiate between function arguments. However, it should leave the **choice of colors** to the **theme** chosen by package users.

As it stands now, the default color chosen for arguments (blue) is completely out-of-place in my `rusty-dark-syntax` theme. I was able to work around this by defining very specific style overrides with `!important` css rule. However, this is bad for both of us:

- To my knowledge, no other language plugin is forcing colors, and I don't like to create a bunch of exceptions for the Rust language;
- If I create exceptions, the settings in your package _will not work_;
- Users who used other theme before installing your package will get sub-par experience when editing rust files, and may even conclude that Rust is broken on Atom. They usually don't know how syntax coloring works.

There is additional issue of `atom-language-rust` having different styles than `language-rust`, however it is not as bad as this situation of broken themes.

**Comment by miqid** [comment]
Hello @Nercury and thanks for your feedback.

I'm well aware that most popular grammar packages for Atom do not interfere with applying styles across tokenised language constructs. I've not documented it clearly, but it is possible to suppress all styles introduced by `atom-language-rust` simply by unchecking the _Additional Styles_ setting for the package. With this setting disabled, any syntax theme that is willing to cover all the grammar scope names that I've introduced should be able to provide syntax highlighting however it wants.

I originally started development on the providing custom styles along with grammar scopes because all of the default Atom UI and syntax themes don't contain grammar scopes that are specific to the Rust language. For example, a quick survey of the [language styles](https://github.com/atom/one-dark-syntax/blob/master/styles/language.less) for the default Atom syntax theme `one-dark-syntax` reveals that elements like lifetimes and macros aren't specified. While the (unsaid) convention appears to be submitting a language specific LESS file to syntax themes, I don't agree with it. In short, every single available Atom theme would need to provide styles with grammar scopes from `atom-language-rust` in mind. Rather than have the syntax theme developers out there carry the burden for Rust-specific constructs, I believe it's far more convenient and manageable for it to be available along with the grammar package, with the ability to opt out (i.e. _Additional Styles_ setting).

There's still a fair chunk of work I plan on doing for `atom-language-rust`. Consequently, a lot of the grammar scopes will likely change, or even disappear.

Bottom line is, if you feel that `atom-language-rust` is unsuitable for use with `rusty-dark-syntax`, direct users to `language-rust`. There's always choice available. :smile:

If you'd like to discuss in greater detail, I can be found on the Atom Slack channels.

**Comment by Nercury** [comment]
> I'm well aware that most popular grammar packages for Atom do not interfere with applying styles across tokenised language constructs.

> While the (unsaid) convention appears to be submitting a language specific LESS file to syntax themes, I don't agree with it.

IMO, going against (even unsaid) convention will not be good for Rust in the long run. Have you tried working together with `rust-language` authors? Your package now is the first in the list of suggested packages for Rust, and it will definitely have an impact, one way or another. It would be best for this impact to make for smoother experience with minimal need to choose between very similar solutions.
